### PR TITLE
Close popover on scroll

### LINKED ISSUE #5139: Popover does not work inside scrollable div
### Provide a general summary of the issue here

Popover component stays open and doesn't update its position when it's put inside a scrollable div.

The video below demonstrate what I meant. I'm using ComboBox element here, which uses Popover under the hood. Here's the [CodeSandBox](https://codesandbox.io/s/bold-hodgkin-g7324l?file=/src/App.js) link.

https://github.com/adobe/react-spectrum/assets/42168740/94b0369a-ff41-4fa7-bcd3-78a45dd5bb85

### 🤔 Expected Behavior?

The Popover should close, or its position should be updated when the parent div is scrolled.

### 😯 Current Behavior

The Popover neither close, nor update its position when the parent div is scrolled.

### 💁 Possible Solution

LFDanLu suggested updating this line to achieve the desired behavior.

https://github.com/adobe/react-spectrum/discussions/5132#discussioncomment-7117165

### 🔦 Context

_No response_

### 🖥️ Steps to Reproduce

1. Put React Spectrum's ComboBox within a scrollable div.
2. Open the ComboBox's Popover.
3. Scroll the parent div.

[CodeSandBox](https://codesandbox.io/s/bold-hodgkin-g7324l?file=/src/App.js)

### Version

latest

### What browsers are you seeing the problem on?

Firefox

### If other, please specify.

_No response_

### What operating system are you using?

MacOS

### 🧢 Your Company/Team

_No response_

### 🕷 Tracking Issue

_No response_

### PR DESCRIPTION
**Pull request by boutahlilsoufiane**
Closes #5139

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by LFDanLu** [comment]
GET_BUILD

**Review by LFDanLu — COMMENTED** [review]
Thank you for updating this and for figuring out my messed up logic from https://github.com/adobe/react-spectrum/discussions/5132#discussioncomment-7117165, just realized it had a extraneous negation sorry about that.

Since we don't have a test suite for `usePopover` directly, would you mind adding a test to the Combobox test suites ([RAC](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/test/ComboBox.test.js) and [RSP](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/combobox/test/ComboBox.test.js)) and maybe one for [Popover](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/test/Popover.test.js)? I imagine it could be something like this (RAC test example):

```
  it.only('should ', async () => {
    let {getByRole} = render(
        <ComboBox isRequired>
          <Label>Favorite Animal</Label>
          <Input />
          <Button />
          <FieldError />
          <Popover>
            <ListBox>
              <ListBoxItem id="1">Cat</ListBoxItem>
              <ListBoxItem id="2">Dog</ListBoxItem>
              <ListBoxItem id="3">Kangaroo</ListBoxItem>
            </ListBox>
          </Popover>
        </ComboBox>
    );

    let button = getByRole('button');
    await user.click(button);
    let listbox = getByRole('listbox');
    expect(listbox).toBeInTheDocument();
    fireEvent.scroll(document.body);
    expect(listbox).not.toBeInTheDocument();
  });
```

**Comment by boutahlilsoufiane** [comment]
Yes ofc, I will add the test suite.

**Comment by snowystinger** [comment]
GET_BUILD

File: packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx [hunk 1]
```diff
@@ -980,3 +984,39 @@ function ComboBoxWithinDialog(props) {
...
+    >
+      <div
+        style={{
+          paddingTop: "50vh",
+          height: "150vh",
+          display: "flex",
+          justifyContent: "center",
+        }}
+      >
+        <ComboBox container={root} label="Combobox" {...props}>
```
[the remark this answers is not in the document]
Comment (boutahlilsoufiane) [reply packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx:1012]: Thanks for catching this, I have to remove it.

**Comment by boutahlilsoufiane** [comment]
Hi @LFDanLu , I removed the story, isn't necessary to add another test to cover the main issue in #5139?

**Comment by LFDanLu** [comment]
@boutahlilsoufiane would be nice to have, but I won't make it a blocker for the PR since we have a test for the useOverlayPosition hook that tests that onClose fires when the scroll happens in the scrollable container. The test that covers body scroll for ComboBox should be sufficient to confirm that useOverlayPosition's onClose is being properly provided for the ComboBox case

**Review by LFDanLu — APPROVED** [review]

**Comment by boutahlilsoufiane** [comment]
Hi @LFDanLu. Could you please tell how do you fix linting issues?

**Comment by LFDanLu** [comment]
@boutahlilsoufiane Sure! I ran `yarn lint` locally which yielded the following error (below taken from the build since I've closed my terminal window that I originally ran lint on): 
```
 /home/circleci/react-spectrum/packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx
[1]   1007:2  error  Newline required at end of file but not found  eol-last
```
The above indicates that we just needed a newline in the ComboBox stories file so I went ahead and added that.

There are a bunch of other lint warnings that we are using to track progress on a different issue, so it is probably hard to find the actual errors when running lint, sorry about that haha.

**Review by snowystinger — APPROVED** [review]

**Pull request merged**